File: src/DatabaseLib/models/ValueInfo.cs
using gov.sandia.sld.common.configuration;
using Newtonsoft.Json;
using System;

namespace gov.sandia.sld.common.db.models
{
    public class ValueInfo
    {
        public int deviceID { get; set; }
        public ECollectorType collectorType { get; set; }
        public string value { get; set; }
        public DateTimeOffset timestamp { get; set; }
        [JsonIgnore]
        public bool IsValid { get { return string.IsNullOrEmpty(value) == false && timestamp != DateTimeOffset.MinValue; } }

        public ValueInfo()
        {
            deviceID = -1;
            collectorType = ECollectorType.Unknown;
            value = string.Empty;
            timestamp = DateTimeOffset.MinValue;
        }
    }
}

File: src/COMMONWeb/app/classes/valueinfo.ts
import { ECollectorType } from "../enums/collectortype.enum";
import { IValueInfo } from "./ivalueinfo";

// See Models.cs
export class ValueInfo {
    deviceID: number;
    collectorType: ECollectorType;
    value: string;
    timestamp: Date;

    constructor(ivi: IValueInfo) {
        this.deviceID = ivi.deviceID;
        this.collectorType = ivi.collectorType;
        this.value = ivi.value;
        this.timestamp = new Date(ivi.timestamp);
    }

    public parseValue(): string[] {
        var value: string[] = [];
        if (this.value)
            value = JSON.parse(this.value)['Value'];
        return value;
    }
}
